### PR TITLE
Resolves #2440 Image paste from clipboard support

### LINKED ISSUE #2440: Image paste from clipboard broken
Hi,

just installed myBB on our server. First thing I tested was image paste from the clipboard ... but this seems to be broken. Is this a known issue? I couldn't find it in the forum or on github.
Short description:
Paste image works in the editor, but displays as source code in the preview and in the final posted version

If its not a known issue, I'm happy to provide more details.

Jan

Original thread: [Image paste from clipboard broken](http://community.mybb.com/thread-194877.html)

### PR DESCRIPTION
**Pull request by effone**
Resolves #2440

Image paste from clipboard support added.

### CODE REVIEW

File: inc/class_parser.php [hunk 1]
```diff
@@ -1332,6 +1343,17 @@ function mycode_parse_img_disabled_callback4($matches)
...
 
+	/**
+	 * Parses IMG MyCode disabled.
+	 *
+	 * @return string Raw image code notice.
+	 */
+	function mycode_parse_img_disabled_callback5()
+	{
+		global $lang;
+		return $lang->posted_image_uri;
```
Comment (euantorano) [inline inc/class_parser.php]: Shouldn't this behave the same as the other disabled images by calling `mycode_parse_img_disabled`?

**Comment by dvz** [comment]
Do we want inline image support in the long run (rather than handling them as attachments client-side when pasted/dropped)? Best to go over this before adding support for formats that may be dropped later @mybb/developers

**Comment by euantorano** [comment]
IMO this is what should happen:

1. Image pasted/dropped in to editor
2. JS detects this, and starts an AJAX request to create an attachment
3. If attachment added, atatchment is inserted in post/thread/PM (note: PMs don't support attachments in 1.8...)

In the long run, base 64 content for what could be large (several MB) image files inside the posts table would only cause headaches.

However, implementing the above wouldn't be finished by 1.8.22 unless we want to delay the release by several weeks.

I wouldn't mind stripping out the base 64 support for now and creating a new issue for 1.8.23 to implement the above flow. Or we can leave it as is for 1.8.22 and implement the above for 1.8.23 - only problem will be any posts created between 1.8.22 and 1.8.23 could have inline base 64 image data.

**Comment by dvz** [comment]
Yes, we could revert this one to avoid potential backtracking on MyCode support (individual forums could still install a plugin for this, which should be simple to create).

**Comment by euantorano** [comment]
I think that would be for the best for now.

**Pull request merged**